src/csm/Extensions/LoadingExtension.cs

```diff
@@ -51,6 +53,10 @@ public override void OnLevelLoaded(LoadMode mode)
                 panel.DisplayReleaseNotes();
                 CSM.Settings.LastSeenReleaseNotes.value = strVersion;
             }
+
+            //Registers all other mods which implement the API
```

Comment:
I think this should happen earlier, if a mod wants to send data on level load, this may be too late. We also might need this already in the main menu to check if the server and client have the same mods (supporting CSM) installed.